pkg/operation/botanist/addons.go

```diff
@@ -284,9 +251,8 @@ func (b *Botanist) generateOptionalAddonsChart(_ context.Context) (*chartrendere
 	}
 
 	return b.ShootClientSet.ChartRenderer().Render(filepath.Join(charts.Path, "shoot-addons"), "addons", metav1.NamespaceSystem, map[string]interface{}{
-		"global":               global,
-		"kubernetes-dashboard": kubernetesDashboard,
```

Comment:
I think you have to do 
```suggestion
		"kubernetes-dashboard": common.GenerateAddonConfig(nil, true),
```
for now, and remove this later.
Otherwise the kubernetes-dashboard resources will be recreated if it's missing from `shoot-core` MR.